scripts/name_giver.py

```diff
@@ -0,0 +1,73 @@
+import os
+import argparse
+import uuid
+
+"""
+Give unique names to all image files in a given folder (and its subfolders).
+"""
+
+IMG_EXTENSIONS = ['.jpg', '.JPG', '.jpeg', '.JPEG', '.png', '.PNG']
+
+
+def generate_unique_name(include_index=False, index=None):
+    """
+    Generate a unique name with (for convenience in cleaning data for training)
+    or without index before name.
+    """
+
+    unique_name = uuid.uuid4()
+
+    if include_index:
+        given_name = str(index) + '__' + str(unique_name)
+        index += 1
+    else:
+        given_name = str(unique_name)
+
+    return given_name
+
+
+def get_all_files(input_directory):
+    """
+    Return all files to be renamed (to a unique name) as a list of strings.
+    """
+
+    filenames = []
+
+    for dir_name, subdir_list, file_list in os.walk(input_directory):
+        print('Found directory: {}'.format(dir_name))
+
+        for filename in file_list:
+            ext = '.' + filename.split('.')[-1]
+            if ext in IMG_EXTENSIONS:
+                filenames.append(os.path.join(dir_name, filename))
+
+    return filenames
+
+
+def get_file_extension(filename):
+    return filename.split('.')[-1]
```

Comment:
these are more util functions than they are script specific functions. They're fine here for now but if they're used anywhere else in the program they should be placed in a separate file